lib/index.d.ts

```diff
@@ -0,0 +1,600 @@
+            idValue: string,
+            priceToEstimateFees: {
+                listingPrice: {
+                    amount: string,
+                    currencyCode: string,
+                },
+                shipping: {
+                    amount: string,
+                    currencyCode: string,
+                },
+            },
+        },
+        status: 'ClientError' | 'ServerError' | string, // TODO: not sure what a value here is with no error
+        // totalFees only if no error
+        totalFees?: {
+            amount: string,
+            currencyCode: string,
+        },
+        // time only if no error
+        time?: string,
+        // detail only if no error, need to fill this in
+        detail?: Array<any>,
+        error?: {
+            code: string,
+            message: string,
+            type: 'Sender' | 'Receiver',
+        },
+    },
+};
+
+type NextTokenParams = { NextToken: string };
+
+type RequestReportParams = {
+    ReportType: string,
+    ReportProcessingStatus?: string,
+    EndDate?: string,
+    Scheduled?: boolean,
+    ReportRequestId?: string,
+    SubmittedDate?: string,
+    StartDate?: string,
+};
+type RequestReportReturn = any;
+
+type GetReportRequestListParams = {
+    ReportRequestIdList?: Array<string>,
+    ReportTypeList?: string,
+    ReportProcessingStatusList?: string,
+    MaxCount?: number,
+    RequestedFromDate?: Date,
+    RequestedToDate?: Date,
+};
+type GetReportRequestListReturn = {
+    nextToken?: string,
+    reportRequestList: Array<{
+        ReportType: string,
+        ReportProcessingStatus: string,
+        EndDate: string,
+        Scheduled: boolean,
+        ReportRequestId: string,
+        StartedProcessingDate: string,
+        StartDate: string,
+        CompletedDate: string,
+        GeneratedReportId: string,
+    }>,
+};
+
+type GetReportParams = { ReportId: string };
+type GetReportReturn = any;
+
+type GetReportListParams = {
+    MaxCount?: number,
+    ReportTypeList?: Array<string>,
+    Acknowledged?: boolean,
+    ReportRequestIdList?: Array<string>,
+    AvailableFromDate?: Date,
+    AvailableToDate?: Date,
+};
+type GetReportListReturn = any;
+
+type RequestAndDownloadReportReturn = any;
+
+type ManageReportScheduleParams = {
+    ReportType: string,
+    Schedule: any,
+    ScheduleDate?: Date,
+};
+type ManageReportScheduleReturn = any;
+
+type UpdateReportAcknowledgementsParams = {
+    ReportIdList: Array<string>,
+    Acknowledged?: boolean,
+};
+type UpdateReportAcknowledgementsReturn = any;
+
+type GetReportScheduleListParams = { ReportTypeList?: Array<string> };
+type GetReportScheduleListReturn = any;
+
+type IdTypes =
+    | 'ASIN'
+    | 'GCID'
+    | 'SellerSKU'
+    | 'UPC'
+    | 'EAN'
+    | 'ISBN'
+    | 'JAN';
+
+type AdvIdTypes =
+    | 'asin'
+    | 'gcid'
+    | 'sellersku'
+    | 'upc'
+    | 'ean'
+    | 'isbn'
+    | 'jan';
+
+declare class MwsAdvanced {
+    static constants: {
```

Comment:
Can we get rid of static? Reasons: 
- It clutters our code base a lot
- Is error prone since binding methods is easily forgotten
- Although it's a breaking change, this change is very easy to incorporate in existing code bases that make use of static. For example, these users can do the following:

1. Create a static file:

```{js}
import MwsAdvanced from "@ericblade/mws-advanced"

export default new MwsAdvanced({ ... })
```

2. Import MwsAdvanded from that file and start using it by updating method calls.

Reply:
yep, will do, after this task.

Reply:
https://github.com/ericblade/mws-advanced/issues/172